src/lammpsio/snapshot.py

```diff
@@ -91,22 +91,77 @@ def from_hoomd_gsd(cls, frame):
         snap.charge = frame.particles.charge
         snap.mass = frame.particles.mass
 
-        snap.molecule = frame.particles.body + 1
+        if frame.particles.body is not None:
+            snap.molecule = frame.particles.body + 1
         if numpy.any(snap.molecule < 0):
             warnings.warn("Some molecule IDs are negative, remapping needed.")
 
-        if (
-            frame.bonds.N > 0
-            or frame.angles.N > 0
-            or frame.dihedrals.N > 0
-            or frame.impropers.N > 0
-        ):
-            warnings.warn("Conversion of topology from gsd is not supported")
-        type_map = {typeid + 1: i for typeid, i in enumerate(frame.particles.types)}
+        if frame.bonds.N != 0:
+            snap.bonds = Bonds(N=frame.bonds.N)
+            snap.bonds.id = frame.bonds.typeid
+            snap.bonds.members = frame.bonds.group + 1
```

Comment:
I would move the labelmap logic for each of these bonding types up here, and only assign the labels if they exist. If there are bonds in the GSD file, there has to be at least one type stored.